Makefile

```diff
@@ -160,10 +164,11 @@ test: system
 toolchain:
 	@cd toolchain; ./toolchain-build.sh
 
+ASM_OBJS = kernel/boot.o kernel/gdt.o kernel/idt.o kernel/irq.o kernel/isr.o kernel/task.o kernel/tss.o kernel/user.o
```

Comment:
I know you're trying to force an ordering on these so `boot` is first, but I'd much rather see this automatically generated like the other object lists. You're also duplicating effort from where you add `$(filter-out kernel/symbols.o,$(patsubst %.S,%.o,$(wildcard kernel/*/*.S)))` to `KERNEL_OBJS`.